cmd2/cmd2.py

```diff
@@ -659,6 +668,43 @@ def register_command_set(self, cmdset: CommandSet) -> None:
             cmdset.on_unregistered()
             raise
 
+    def _build_parser(self, parent: Union['Cmd', CommandSet], parser_builder: Any) -> Optional[argparse.ArgumentParser]:
```

Comment:
parser_builder shouldn't be Union[argparse.ArgumentParser, Callable] ?

Reply:
No, the point is to always result in an instance of a parser after this. either by deep-copying the provided one or calling the builder.

Reply:
oh wait, misread your comment.

Reply:
Ah, this got messy because `staticmethod` and `classmethod` are not considered callable for some reason and there's some complicated mess where mypy wants to treat them like a generic type but at runtime this causes a python parsing error and fails..